### PR TITLE
Added docs website scaffolding

### PR DESCRIPTION
**Pull request by ellistarn**
**Issue, if available:**

**Description of changes:**
Example: https://ellistarn.github.io/karpenter/docs/

- Everything in /docs is now autogenerated by code in /website
- Moved a few things around to make the above true.
- Helm hosted from awslabs.github.com/charts
- Docs hosted from awslabs.github.com/docs

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Contributing
+# Developer Guide
 
 ## Dependencies
 
-The following tools are required for doing development on Karpenter.
+The following tools are required for developing the Karpenter operator.
```
Comment (bwagner5) [inline DEVELOPER_GUIDE.md:5]: calling it an operator now?

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ The following tools are required for doing development on Karpenter.
 | [helm](https://helm.sh/docs/intro/install/)                        |          | `brew install helm`    |
 | Other tools                                                        |          | `make toolchain`       |
 
+The following tools are required to developing the Karpenter website..
```
Comment (bwagner5) [inline DEVELOPER_GUIDE.md:14]: "to develop" or "for developing"

File: docs/404.html [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/favicon-16x16.png" sizes="16x16">
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/favicon-32x32.png" sizes="32x32">
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/android-36x36.png" sizes="36x36">
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/android-48x48.png" sizes="48x48">
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/android-72x72.png" sizes="72x72">
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/android-96x96.png" sizes="96x96">
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/android-144x144.png" sizes="144x144">
+<link rel="icon" type="image/png" href="/karpenter/docs/favicons/android-192x192.png" sizes="192x192">
+
+<title>404 Page not found | My New Hugo Site</title>
```
Comment (bwagner5) [inline docs/404.html:20]: My New Hugo Site  to something else ?

File: docs/404.html [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+<meta name="description" content=""><meta property="og:title" content="404 Page not found" />
+<meta property="og:description" content="" />
+<meta property="og:type" content="website" />
+<meta property="og:url" content="http://ellistarn.github.com/karpenter/docs/404.html" />
```
Comment (bwagner5) [inline docs/404.html:24]: old :D

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (ellistarn) [reply DEVELOPER_GUIDE.md:5]: Operator is a controller+webhook :)

File: docs/404.html (hunk 3 above)
Comment (ellistarn) [reply docs/404.html:20]: This will get retooled by @geoffcline

**Review by bwagner5 — APPROVED** [review]
/lgtm

**Pull request merged**